sprite.go

```diff
@@ -286,75 +279,20 @@ func (p *SpriteImpl) init(
 		if ani.StepDuration == 0 {
 			ani.StepDuration = 0.01
 		}
-		switch ani.AniType {
-		case aniTypeFrame:
-			if ani.From == nil {
-				if ani.FrameFrom != "" {
-					ani.From = ani.FrameFrom
-				} else {
-					log.Panicf("animation key [%s] missing FrameFrom ", key)
-				}
-				if ani.FrameTo != "" {
-					ani.To = ani.FrameTo
-				} else {
-					log.Panicf("animation key [%s] missing FrameTo ", key)
-				}
-			} else {
-				if str, ok := ani.From.(string); ok && str != "" {
-					ani.FrameFrom = ani.From.(string)
-					ani.FrameTo = ani.To.(string)
-				}
-				ani.From, ani.To = p.getFromAnToForAniFrames(ani.From, ani.To)
-			}
-			from, to := p.getFromAnToForAniFrames(ani.From, ani.To)
-			if oldFps == 0 && oldFrameFps != 0 {
-				ani.Fps = float64(oldFrameFps)
-				ani.FrameFps = oldFrameFps
-				ani.Duration = math.Abs(to-from) / ani.Fps
-			} else if oldFps != 0 {
-				ani.Fps = oldFps
-				ani.FrameFps = int(oldFps)
-				ani.Duration = math.Abs(to-from) / ani.Fps
-			} else {
-				ani.Fps = math.Abs(to-from) / ani.Duration
-				ani.FrameFps = int(ani.Fps)
-			}
-		case aniTypeMove:
-		case aniTypeTurn:
-		case aniTypeGlide:
-		default:
-			log.Panicf("unknown animation type [%s] is exist[%d]", key, ani.AniType)
-		}
-		p.animations[key] = ani
-	}
-
-	for key, val := range spriteCfg.MAnimations {
-		_, ok := p.animations[key]
-		if ok {
-			log.Panicf("animation key [%s] is exist", key)
-		}
-		var ani = val
-		ani.AniType = aniTypeMove
-		if ani.Fps == 0 {
-			ani.Fps = 25
-		}
+		from, to := p.getFromAnToForAniFrames(ani.FrameFrom, ani.FrameTo)
+		ani.IFrameFrom, ani.IFrameTo = int(from), int(to)
+		ani.Speed = 1
+		ani.Duration = math.Abs(float64(ani.IFrameFrom-ani.IFrameTo)) / float64(ani.FrameFps)
```

Comment:
There's an edge detail here.

With `frameFrom: 0` and `frameTo: 1`, there should be two frames (`costumes[0]` and `costumes[1]`), correct?

If `frameFps: 1`, should the animation duration be 1 second or 2 seconds?